code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -336,6 +337,28 @@ func (zcn *ZCNSmartContract) UpdateAuthorizerConfig(
 		return "", common.NewError(code, err.Error())
 	}
 
+	sp, err := zcn.getStakePool(in.ID, ctx)
+	if err != nil {
+		if err == util.ErrValueNotPresent {
+			msg := fmt.Sprintf("update not allowed due to missing stakepool details for (authorizerID: %v), err: %v", authorizer.ID, err)
+			err = common.NewError(code, msg)
+			Logger.Error("updating settings", zap.Error(err))
+			return "", err
+		}
+		msg := fmt.Sprintf("unexpected error for (authorizerID: %v), err: %v", authorizer.ID, err)
+		err = common.NewError(code, msg)
+		Logger.Error("updating settings", zap.Error(err))
+		return "", err
+	}
+
+	// Do not allow authorizer update in case of mismatch
+	if sp.Settings.DelegateWallet != t.ClientID {
+		msg := fmt.Sprintf("mismatched delegate wallet id for (authorizerID: %v)", authorizer.ID)
```

Comment:
we need to change error text to something like 
``unauthorized access - only managing wallet can access ``
please check this, I suggest to add AuthorizeWithDelegate and make it in a similar way https://github.com/0chain/0chain/blob/79a0fd029fa398b87be187912d9345ce25635709/code/go/0chain.net/chaincore/smartcontractinterface/auth.go